<!-- jplaced@unizar.es
 2022, Universidad de Zaragoza -->

<!-- Launch file for the dopt_exploration -->
<launch>
  <param name="namespace_init_count" value="1"/>
  <!-- This parameter controls the growth rate of the local RRT that is used in
  the detection of frontier points, the unit is in meters. This parameter should
  be set according to the map size, a very large value will cause the tree to
  grow faster and hence detect frontier points faster, but a large growth rate
  also implies that the tree will be missing small corners in the map. -->
  <arg name="eta" value="3.0"/>
  <!-- Same for the global RRT detector. -->
  <arg name="Geta" value="300.0"/>
  <param name="use_sim_time" value="true" />

  <!-- <group ns="/robot_1"> -->

  <node pkg="graph_d_exploration" type="opencv_detector.py" name="opencv_detector" output="screen">
    <param name="map_topic" value="/map"/>
    <param name="rate" value="25.0" />
  </node>

  <node pkg="graph_d_exploration" type="d_opt_global_rrt_detector" name="global_detector" output="screen">
    <param name="eta" value="$(arg Geta)"/>
    <param name="map_topic" value="/map"/>
    <param name="rate" value="25.0" />
  </node>

  <node pkg="graph_d_exploration" type="d_opt_local_rrt_detector" name="local_detector" output="screen">
    <param name="eta" value="$(arg eta)"/>
    <param name="map_topic" value="/map"/>
    <param name="robot_frame" value="base_link"/>
    <param name="rate" value="25.0" />
  </node>

  <node pkg="graph_d_exploration" type="filter.py" name="filter" output="screen" respawn="true" respawn_delay="3">
    <param name="map_topic" value="/map"/>
    <param name="costmap_clearing_threshold" value="70"/>
    <param name="goals_topic" value="/detected_points"/>
    <param name="namespace" value="/robot_"/>
    <param name="n_robots" value="1"/>
    <param name="rate" value="25.0"/>
    <param name="info_radius" value="1"/>
  </node>

  <node pkg="graph_d_exploration" type="initializer.py" name="point_init" output="screen">
    <param name="rate" value="1"/>
  </node>

  <node pkg="graph_d_exploration" type="controller_graphD.py" name="assigner" output="screen">
    <param name="map_topic" value="/map"/>
    <param name="global_frame" value="map"/>
    <param name="frontiers_topic" value="/filtered_points"/>
    <param name="namespace" value="/robot_"/>
    <param name="n_robots" value="1"/>
    <param name="delay_after_assignement" value="0.1"/>
    <param name="rate" value="1"/>
    <param name="info_radius" value="1.5"/>
  </node>

</launch>
